### PR TITLE
Fix SonarQube issues

### PR DESCRIPTION
**Pull request by peterbae**

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java [hunk 1]
```diff
@@ -4880,7 +4880,7 @@ else if (DataTypes.UNKNOWN_STREAM_LENGTH == dataLength)
                 isShortValue = columnPair.getValue().precision <= DataTypes.SHORT_VARTYPE_MAX_BYTES;
                 isNull = (null == currentObject);
                 if (currentObject instanceof String)
```
Comment (cheenamalhotra) [inline src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java]: should we not check isNull here ?

File: src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java [hunk 2]
```diff
@@ -4880,7 +4880,7 @@ else if (DataTypes.UNKNOWN_STREAM_LENGTH == dataLength)
...
-                    dataLength = isNull ? 0 : (ParameterUtils.HexToBin(currentObject.toString())).length;
+                    dataLength = ParameterUtils.HexToBin(currentObject.toString()).length;
                 else
                     dataLength = isNull ? 0 : ((byte[]) currentObject).length;
```
Comment (cheenamalhotra) [inline src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java]: If isNull is checked above, this check can go away too

Comment (peterbae) [reply src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java]: i think this one stays?

Comment (cheenamalhotra) [reply src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java]: Also, this method has 2 object instances `currentObject` and `currentObjectStringValue` that are compared to null over and over again, which can be done in the beginning of the method and a boolean instance for each can be used everywhere we need to check for null. SqlVariant has no null value check for `currentObject` that should be handled too..

File: src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java (hunk 1 above)
Comment (peterbae) [reply src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java]: no, if currentObject is an instance of String, it cannot be null, so we dont have to check isNull here.

Comment (cheenamalhotra) [reply src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java]: Turns out `null instanceOf <Class>` does not error out, so we can avoid null check

File: src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java (hunk 2 above)
Comment (cheenamalhotra) [reply src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java]: As discussed, we'll keep that change for later

**Pull request merged**